### ISSUE TITLE
Synced folders should support syncing multiple sources to same destination

### ISSUE THREAD

**Issue by jhalterman** [mitchellh/vagrant#8518]
### Expected behavior
Vagrant's synced folder feature, with rsync in particular, should support syncing multiple sources to the same destination. With rsync this can normally be accomplished with:

```
rsync -a source1/ source2/ source3/ destination/
```

### Actual behavior
When trying to approximate this with Vagrant with something like:

```
current.vm.synced_folder source1 dest
current.vm.synced_folder source2 dest
current.vm.synced_folder source3 dest
```

...only the last `synced_folder` statement is applied since Vagrant appears to key off of the destination folder.

**Comment by chrisroberts** [comment]
This sounds like a reasonable idea, but this has low chance of being a priority for us anytime soon, so we'd rather close this issue than have it stagnate into the foreseeable future. The best way to get this implemented would be to submit a PR. If you do so, we'd welcome this work. If you want to discuss how to go about doing this, I'd be happy to help with that.

Again, thank you for suggesting this and it _is_ a reasonable idea. But, since this is a feature request, it requires some work to add it to the project and we don't plan on doing that soon. But as this is also an open source project we welcome contributions. Thanks!

**Comment by jhalterman** [comment]
Thanks Chris. Without hacking on Vagrant itself, do you think a quicker path forward (workaround) for me would be to just write a quick plugin that runs rsync as I wish?

**Comment by chrisroberts** [comment]
Hey @jhalterman! The plugin approach would definitely be quicker and have the added bonus of being specific to your use case.

**Comment by dinbrca** [comment]
For anyone having the same problem..

I needed to sync to source folder to same destination and had the same problem.
I solved the problem by having 2 destinations on the vagrant machine.. one of them is real folder and another one is a symbolic link to the real folder.

so let's say I need to sync 3 sources to the same folder on guest called 'scripts'
I would have 3 folders on the guest:
- scripts
- scripts_link1
- scripts_link2

and on the `Vagrantfile` I would have the following lines:
```
    config.vm.synced_folder 'path/to/source1', "/srv/scripts", type: "rsync", rsync__args: ["-r"], rsync__verbose: true
    config.vm.synced_folder 'path/to/source2', "/srv/scripts_link1", type: "rsync", rsync__args: ["-r"], rsync__verbose: true
    config.vm.synced_folder 'path/to/source3', "/srv/scripts_link2", type: "rsync", rsync__args: ["-r"], rsync__verbose: true
```

The problem is of course in vagrant decision regarding the path.
Notice the -r argument in the rsync_args or else rsync will throw "skipping main directory" (https://serverfault.com/questions/363555/why-is-rsync-skipping-the-main-directory).
